### PR TITLE
refs [TASK][33689] init ui detail

### PR DESCRIPTION
**Pull request by phong10101998**
## Related Tickets
- [#Task 33689: Content](https://edu-redmine.sun-asterisk.vn/issues/33689)
## WHAT
- Create UI Fragment Detail (Image, Poster, title, time cook, viewpager step, recylerview similar reicpe)
- Show list step by viewpager2
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <resources>
     <string name="app_name">BookCook</string>
     <string name="title_header">cookbook.com</string>
-    <string name="step">Steps :</string>
+    <string name="step">Step : %s</string>
+    <string name="step_detail">" %s "</string>
```
Comment (tuanlvt-1117) [inline app/src/main/res/values/strings.xml]: cái này thì cần gì đưa vào string e?

File: app/src/main/res/drawable-v24/bg_step.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <corners
+        android:bottomLeftRadius="20dp"
+        android:bottomRightRadius="20dp"
+        android:topLeftRadius="20dp"
+        android:topRightRadius="20dp" />
+    <solid android:color="#F2F2F2"/>
+</shape>
```
Comment (tuanlvt-1117) [inline app/src/main/res/drawable-v24/bg_step.xml]:
```suggestion
</shape>

```

File: app/src/main/res/layout/fragment_detail_recipe.xml [hunk 3]
```diff
@@ -1,8 +1,187 @@
...
     android:focusable="true">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <ImageView
+            android:id="@+id/imgFavourite"
+            android:layout_width="@dimen/dp_30"
+            android:layout_height="@dimen/dp_30"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_detail_recipe.xml:18]: chia tỉ lệ ảnh trong ConstraintLayout. Hạn chế việc set cứng

File: app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt [hunk 4]
```diff
@@ -5,26 +5,75 @@ import androidx.fragment.app.Fragment
...
+        textTime.text = context?.getString(R.string.ready_in_minute, recipes.timeCook.toString())
+        recipes.step?.run {
+            initStep(this)
+        }
     }
 
     override fun onError(exception: Exception?) {
+        if (exception == null) Toast.makeText(context, Constant.NULL_STRING, Toast.LENGTH_SHORT)
+            .show()
+        else Toast.makeText(context, exception.message, Toast.LENGTH_SHORT).show()
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt]:
```suggestion
       Toast.makeText(context, exception?.message?.toString(), Toast.LENGTH_SHORT).show()
```

Comment (huytq-2422) [reply app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt]: tuy nhiên nếu exception lỗi nó sẽ Toast lên null, không hợp lý lắm

File: app/src/main/res/layout/fragment_detail_recipe.xml (hunk 3 above)
Comment (phong10101998) [reply app/src/main/res/layout/fragment_detail_recipe.xml:18]: cái này là icon thôi anh, em nghĩ nó cũng tương tự như Button nên em set cứng

**Review by tuanlvt-1117 — APPROVED** [review]

File: app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt [hunk 5]
```diff
@@ -5,26 +5,73 @@ import androidx.fragment.app.Fragment
...
+
     override fun getRecipeDetailSuccess(recipes: RecipeDetail) {
+        Glide.with(this)
+            .load(recipes.image)
+            .into(imageDish)
+        textTitleDish.text = recipes.title
+        textTime.text = context?.getString(R.string.ready_in_minute, recipes.timeCook.toString())
+        recipes.step?.run {
+            initStep(this)
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt:52]:
```suggestion
        recipes.step?.let(::initStep)
```

File: app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt [hunk 6]
```diff
@@ -5,26 +5,73 @@ import androidx.fragment.app.Fragment
...
     }
 
     override fun onError(exception: Exception?) {
+        Toast.makeText(context, exception?.message, Toast.LENGTH_SHORT).show()
+    }
+
+    private fun initPresenter() {
+        detailPresenter.apply {
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/cookbook/screen/detail/DetailRecipeFragment.kt:60]: image

**Pull request merged**